test/extended/apiserver/apply.go

```diff
@@ -172,3 +181,16 @@ func createResource(
 
 	return resourceClient, &unstructuredObj, testNamespace
 }
+
+// waitForSATokens waits for the kube-controller-manager to populate the SA from the unstructured object
+// to be populated with a secret defining its tokens
+func waitForSATokens(saClient corev1client.ServiceAccountInterface, unstructuredObj map[string]interface{}) {
+	name, _, err := unstructured.NestedString(unstructuredObj, "metadata", "name")
+	o.Expect(err).NotTo(o.HaveOccurred())
+	err = wait.PollImmediate(time.Second, 15*time.Second, func() (bool, error) {
+		sa, err := saClient.Get(context.Background(), name, metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
```

Comment:
I'm surprised to see a hard assertion in a polling loop. Is it your intention that a transient failure fail the test vs logging the error and returning false?

Reply:
You're right, given the wait is only 15 seconds even if the error does not go away there's no need to fail early.

Fixed now.